browser/browser_test.go

```diff
@@ -0,0 +1,340 @@
+package browser
+
+import (
+	// "gopkg.in/vmihailenco/msgpack.v2"
+	// "io"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var outChan chan interface{}
+
+func TestMain(m *testing.M) {
+	outChan = make(chan interface{})
+	os.Exit(m.Run())
+}
+
+func TestList(t *testing.T) {
+	var res *ResultSet = nil
+	temp, err := ioutil.TempDir("", "list")
+	FailNotNil(err, t)
+	defer os.Remove(temp)
+	// Make two directories in temp
+	err = os.Mkdir(filepath.Join(temp, "a"), 0777)
+	FailNotNil(err, t)
+	err = os.Mkdir(filepath.Join(temp, "b"), 0777)
+	FailNotNil(err, t)
+	go List("test", outChan, temp)
+	res = (<-outChan).(*ResultSet)
+	if res.Err != "" {
+		t.Fail()
+	}
+	if len(res.Files) != 2 {
+		t.Fail()
+	}
+}
+
+func TestListNotExist(t *testing.T) {
+	var res *ResultSet = nil
+	go List("test", outChan, "does/not/exist")
+	res = (<-outChan).(*ResultSet)
+	if res.Err == "" {
+		t.Log("Error should occur.")
+		t.Fail()
+	}
+	t.Log("Error: ", res.Err)
+}
+
+func TestMakeDirectoryAndRemove(t *testing.T) {
+	home, err := ioutil.TempDir("", "MakeAndRemove")
+	FailNotNil(err, t)
+	paths := []string{"/test_folder", "/test_folder/sub_folder"}
+	var res *ResultSet = nil
+
+	defer func() {
+		go Remove("test", outChan, filepath.Join(home, paths[0]))
+		res = (<-outChan).(*ResultSet)
```

Comment:
outChan is common to all tests, which maybe a bad idea...
I did this so that the channel is empty before the next test